sys/include/net/sock/dtls.h

```diff
@@ -0,0 +1,557 @@
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * sock_udp_t udp_sock;
+ * sock_udp_ep_t local = SOCK_IPV6_EP_ANY;
+ * local.port = 12345;
+ * sock_udp_create(&udp_sock, &local_ep, NULL, 0);
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * After that, we set the address of the remote endpoint where the packet is
+ * supposed to be sent to and the port the server is listening to which is
+ * DTLS_DEFAULT_PORT(20220).
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * sock_udp_ep_t remote;
+ * remote.port = DTLS_DEFAULT_PORT;
+ * remote.netif = gnrc_netif_iter(NULL)->pid;   // only if gnrc_netif_numoff == 1
+ * if (!ipv6_addr_from_str((ipv6_addr_t *)remote.addr.ipv6, addr_str)) {
+ *     puts("Error parsing destination address");
+ *     return;
+ * }
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * After the UDP sock is created, we can proceed with creating the DTLS sock.
+ * Before sending the packet, we must first establish a session with the remote
+ * endpoint using @ref sock_dtls_establish_session(). If the handshake is
+ * successfull and the session is established, we can use its `remote` to send
+ * packets to it with @ref sock_dtls_send().
+ *
+ * If error during any of these operation, the session is closed and the sock
+ * is destroyed using @ref sock_dtls_close_session() and
+ * @ref sock_dtls_destroy().
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * res = sock_dtls_create(&dtls_sock, &udp_sock, DTLS_SOCK_CLIENT_TAG, 0);
+ * if (res < 0) {
+ *     puts("Error creating DTLS sock");
+ *     return;
+ * }
+ *
+ * res = sock_dtls_establish_session(&dtls_sock, &remote, &session);
+ * if (res < 0) {
+ *     printf("Error establishing session: %d\n", (int)res);
+ *     goto end;
+ * }
+ *
+ * res = sock_dtls_send(&dtls_sock, &session, data, datalen);
+ * if (res < 0) {
+ *     printf("Error sending DTLS message: %d\n", (int)res);
+ *     goto end;
+ * }
+ *
+ * res = sock_dtls_recv(&dtls_sock, &session, rcv, sizeof(rcv), SOCK_NO_TIMEOUT);
+ * if (res < 0) {
+ *     printf("Error receiving DTLS message: %d\n", (int)res);
+ *     goto end;
+ * }
+ *
+ * end:
+ * puts("Terminating");
+ * sock_dtls_close_session(&dtls_sock, &session);
+ * sock_dtls_destroy(&dtls_sock);
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * @{
+ *
+ * @file
+ * @brief   DTLS sock definitions
+ *
+ * @author  Aiman Ismail <muhammadaimanbin.ismail@haw-hamburg.de>
+ */
+
+#ifndef NET_SOCK_DTLS_H
+#define NET_SOCK_DTLS_H
+
+#include <assert.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <sys/types.h>
+
+#include "net/sock.h"
+#include "net/sock/udp.h"
+#include "net/credman.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Type for a DTLS sock object
+ *
+ * @note    API implementors: `struct sock_dtls` needs to be defined by
+ *          implementation-specific `sock_dtls_types.h`.
+ */
+typedef struct sock_dtls sock_dtls_t;
+
+/**
+ * @brief Information about the established session with
+ *        the remote endpoint. Used when sending and
```

Comment:
Typically a `@brief` is only one sentence. More information can be added to the `@details` section (which is implicitly created for non-specified paragraphs):

```suggestion
 *        the remote endpoint.
 *
 * This session object is used when sending and receiving data to the endpoint.
```

However, this information is somewhat obvious (as you don't do much else with a `sock`) so maybe the last sentence can be dropped completely.

Reply:
Dropped the last sentence.